### ISSUE TITLE
Make it easy for users to add their custom plugins

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes-sigs/kube-scheduler-simulator#8]
Currently, the simulator only supports default plugins. 
But some users may want to try their custom plugins on the simulator.

- Describe in the documentation how to add a custom plugin to the simulator.
- Add some functions(or some option?) to register the custom plugins

/assign

**Comment by warmchang** [comment]
@sanposhiho thank you for this awesome project!

Here is a blank should be trim ('.dockerignore  '), otherwise the git clone don't work on windows.
(a small typo, so no need to open a separate issue to track. :smile: )
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/.dockerignore%20%20

```
william@WIN-MNJRNPMAROQ MINGW64 /d/thehub
$ git clone https://github.com/kubernetes-sigs/kube-scheduler-simulator
Cloning into 'kube-scheduler-simulator'...
remote: Enumerating objects: 209, done.
remote: Counting objects: 100% (209/209), done.
remote: Compressing objects: 100% (155/155), done.
remote: Total 209 (delta 36), reused 196 (delta 33), pack-reused 0
Receiving objects: 100% (209/209), 1.19 MiB | 181.00 KiB/s, done.
Resolving deltas: 100% (36/36), done.
error: invalid path '.dockerignore  '
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'


william@WIN-MNJRNPMAROQ MINGW64 /d/thehub
$
```

**Comment by sanposhiho** [comment]
Hi @warmchang, thanks for reporting. 
I'll fix it soon 🤦‍♂️

**Comment by sanposhiho** [comment]
Aha, sorry. You said about custom scheduler, not custom plugins. 
I think if your custom scheduler has the same interface as the original scheduler, you can try your custom scheduler with the simulator by changing all internal import statements in the simulator to use your scheduler instead of the original scheduler.

**Comment by sanposhiho** [comment]
/close

Close this issue. This issue will be resolved with #35

**Comment by sanposhiho** [comment]
> This issue will be resolved with https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/35

Oh, really? I don't think so. 😅 

/reopen
